src/librustc_ty/instance.rs

```diff
@@ -141,10 +129,13 @@ fn resolve_associated_item<'tcx>(
                 if name == sym::clone {
                     let self_ty = trait_ref.self_ty();
 
-                    // `CloneShim` requires a monomorphic aka concrete type.
-                    if self_ty.needs_subst() {
-                        return None;
-                    }
+                    let span = tcx.def_span(def_id);
```

Comment:
This results in many more `def_span` calls. Could those explain the significantly increased amount of `metadata_decode_entry` calls?

Reply:
It would be cached though, as `def_id` is always the same. So the wins would be tiny.

Reply:
Fixed in 681b386.